From 099f6d9bcd8937276df90c43fad1119f48c29cac Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 12 Sep 2021 17:19:26 +0300
Subject: [PATCH] [SW-55224] gaudi2: arc: add general region defines

Add general region mask address, and also other memory base
offset which are inside general region.

Change-Id: I73a00ae3f6dadbe9bbc535f879251e3daba786b1
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 2a55053..5a35dd5 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -16,11 +16,13 @@
 
 #define HL_SP_SRAM_BASE		0x1000007FF7FE0000ULL
 #define HL_SP_SRAM_SIZE		0x10000
-#define PCIE_FW_SRAM_BASE	0x27FF0000
-#define FW_SRAM_BASE		0x27FD0000
-#define SP_SRAM_BASE		0x27FE0000
-#define ARC_ICCM_BASE		0x27040000
-#define ARC0_ICCM_BASE		0x27000000
+#define SDRAM_BASE		0x10000000
+#define GEN_REGION_BASE		0x20000000
+#define PCIE_FW_SRAM_BASE	(GEN_REGION_BASE + 0x7FF0000)
+#define FW_SRAM_BASE		(GEN_REGION_BASE + 0x7FD0000)
+#define SP_SRAM_BASE		(GEN_REGION_BASE + 0x7FE0000)
+#define ARC_ICCM_BASE		(GEN_REGION_BASE + 0x7040000)
+#define ARC0_ICCM_BASE		(GEN_REGION_BASE + 0x7000000)
 #define SP_SRAM_PREBOOT_OFFSET	0x1000
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
 #define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + 0x8)
@@ -34,7 +36,6 @@
  */
 #define PPBOOT_DCCM_BASE	0x2706402C
 
-#define SDRAM_BASE		0x11000000
 #define HL_SPI_BASE		0x26000000
 #define HL_CFG_BASE		0xF8000000
 #define I2C_SLAVE_DEFAULT_INTR_MASK	0x48FF
